### ISSUE TITLE
#2778 not fixed

### ISSUE THREAD

**Issue by Jedsters** [FrSkyRC/ETHOS-Feedback-Community#2784]
I don't believe that #2778 has been fixed although I should probably caveat that. 
I suspect that it may have been fixed in the software and may work being simulated on a PC, but it doesn't appear to work as expected on my transmitter. This makes me question whether the issue is one of my action lasting multiple cycles rather than 1 so  increasing / decreasing the value by random amounts. 

Create a new test model, create a new VAR, set a VALUE and action e.g. SA up with a value of 0.
Create Actions of LHL and LHR to SUBTRACT and ADD 1.0 

Actioning LHL and LHR increase or decreases the initial value by varying amounts I've seen as little as 3 and as much as 10.
I've tried changing LHL and LHR to SI / SJ with similar results. 

EDIT: I would also question the purpose of having ALWAYS ON as an available action as when this is selected the value just continuously counts up  / down or divides / multiplies. Does this have a purpose?

July27th nightly 23:12:30

Thanks

**Comment by Wolepo** [comment]
Since 3 days the GVAR do not work properly anymore.

**Comment by bsongis-frsky** [comment]
The simulators seem not released correctly

**Comment by bsongis-frsky** [comment]
Reproduced also on Simulator, some values are skipped

**Comment by Wolepo** [comment]
[zaehler2.zip](https://github.com/FrSkyRC/ETHOS-Feedback-Community/files/12213309/zaehler2.zip)

**Comment by bsongis-frsky** [comment]
I am spending time on them, it should be fixed tomorrow morning after the next nightlies

**Comment by bsongis-frsky** [comment]
- limits ** for each value ** honored (it was only honored on final value before)
- edge option fixed

**Comment by Wolepo** [comment]
It is still calculated incorrectly. The value seems to be arbitrary.

**Comment by bsongis-frsky** [comment]
Really? With the latest nightly it should be good, I have checked on X20S. Perhaps you have a different scenario?

**Comment by bsongis-frsky** [comment]
I reopen it until it's 100% fixed

**Comment by Wolepo** [comment]
Hello
Thanks the issue is open again.
Simulator environment: Win11 Hom/64. Simulator is installed in its own folder. Model and sound are in the default folder.
In order.
Step 1:
I have created a new model and the LS and Var1 is expanded step by step.
LS1 is constantly active after starting the transmitter or the new model and saves a defined basic setting (here -96%) with the first line in Var1. The first line is permanently active despite upstream edge and thus permanently blocks all other actions. Line 1 should only be active for a short time, so it is set inversely. Edge seems to have no effect.
Step 2:

LS2 has the Edge function and is active for 0.1s if key SI is pressed between 0.0s and 0.3s. The 2nd line of Var1 is controlled by this and switches one step higher (+12%) each time. Instead of increasing by 12% each time, a multiple of this is calculated.
The downstream mixer also starts with wrong values and limits only at+100%. Is the source wrong here?
Attached the new model.bin and a video
[GVar.zip](https://github.com/FrSkyRC/ETHOS-Feedback-Community/files/12237946/GVar.zip)

[ring ounter.zip](https://github.com/FrSkyRC/ETHOS-Feedback-Community/files/12237920/ring.ounter.zip)

**Comment by bsongis-frsky** [comment]
Wait, the nightlies are not updated!

**Comment by bsongis-frsky** [comment]
I have fixed something and started them manually, waiting for the new bins now...

**Comment by Wolepo** [comment]
The new simulator is still not available

**Comment by bsongis-frsky** [comment]
I am not sure there is a bug there. The range for the input should be 100%. 
For example if you use a Telemetry Value as Input, -100% will be reached when you hit the minimum for your sensor, and +100% when you hit its maximum.
The same applies here, -100% will be reached when you hit the minimum for your Var.
You should test with MAX as Source, and you use the Var for the Weight. It will work as you expect.

**Comment by Wolepo** [comment]
Hello,
yes, with the range limits of +/- 100% the channel 6 has the correct values. Also the LS6 works correctly. But then I wonder for what these range limits should bring a benefit. My understanding is that they are not supposed to be exceeded. So a connected mixer must also get these values 1 to 1. 
I have now set the limits to +/- 97% and a start value of -96% for control. In Var1 shows Value = -96%, but the mixer of CH6 = 98,7%. According to my calculation it would be 100% / 97% (limit) * 96% (actual value) = 98,969(?). Why this calculation?

**Comment by bsongis-frsky** [comment]
I will remove the scaling for Vars now so that you can test it tomorrow after the next nightlies

**Comment by Wolepo** [comment]
Hello,
now everything is as it should be. 
For the functions multiplication and division I can't think of a practical example at the moment, but they will surely come.
In this context I would like to thank you for the good cooperation.
Wolfgang

**Comment by Wolepo** [comment]
Hello,
Sorry, but I was hoping that it is not because of the Vars, but unfortunately it is.
In line 1 of the GVars I have activated [Edge] (Edge transition) and Inverse in the source. Inverse because otherwise the first line is constantly valid, since LS1 is also active and that must not be, because otherwise Values is constantly reset to -96%.
As a counter check I have inserted LS8 and LS9. LS8 is active when SA is in the middle. LS9 is switched active by LS8, but is only active for a short time because of [Edge] (set to 0.3s). Here [Edge] works.
[ring counter.zip](https://github.com/FrSkyRC/ETHOS-Feedback-Community/files/12292501/ring.counter.zip)
Enclosed again the .bin.
